config/config.go

```diff
@@ -370,6 +370,13 @@ func (c *ScrapeConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 			}
 		}
 	}
+
+	// Add index to the static config target groups for unique identification
+	// within scrape pool
+	for i, tg := range c.ServiceDiscoveryConfig.StaticConfigs {
```

Comment:
this could also be done in [targetgroup.UnmarshalYAML()](https://github.com/prometheus/prometheus/blob/25e2d9f152b634d65123ef11342aceb23e66a1a2/discovery/targetgroup/targetgroup.go#L40). This would avoid duplication for handling AlertManager SD.

Reply:
It doesn't work within the target group because the id/index to set for tg.Source is based on the array index which is not known in the targetgroup unmarshal function.

I also wasn't able to remove the duplication using the ServiceDiscoveryConfig unmarshall because it's inlined in ScrapeConfig and AlertmanagerConfig, so its unmarshal function never gets called.

I could move the loop to a function to reduce the duplication a bit, wdyt?

Reply:
Ah right, I misread the code. I'm fine with it then.